src/app/shared/components/addregressionregion/addregressionregion.component.ts

```diff
@@ -493,5 +548,152 @@ export class AddRegressionRegionModal implements OnInit, OnDestroy {
     }, 100);
   }
 
+  /////////////////////// Limitation Section ///////////////////////////  
+
+  public getLimitations(){
+  this._settingsService.getEntities(this.configSettings.nssBaseURL + this.configSettings.regRegionURL + '/' + this.selectedRegRegion.id + '?includeGeometry=true')
+    .subscribe((res) => {
+      this.selectedRegRegion = res;
+      this.limitations = this.selectedRegRegion.limitations;
+    });
+  }
+
+  public addLimitation() {
+    if (this.addRegReg == true) { //Creating new limitation and new regresion region at the same time
+      this.newLimForm.controls['regressionRegionID'].setValue(null);
+    } else{
+      this.newLimForm.controls['regressionRegionID'].setValue(this.selectedRegRegion.id);
+    }
+  } 
+
+  public removeVariable(varIndex) {
+    const control = <FormArray>this.newLimForm.get('variables');
+    control.removeAt(varIndex);
+  }
+
+  public addVariable() {
+    const control = <FormArray>this.newLimForm.get('variables');
+    control.push(this._fb.group({
+      id: new FormControl(null),
+      limitationID: new FormControl(null),
+      variableTypeID: new FormControl(null, Validators.required),
+      unitTypeID: new FormControl(null, Validators.required),
+    }));
+  }
+
+  private cancelCreateLimitaiton() {
+    this.newLimForm.reset();
+    this.addLim = false;
+    this.editFormLimitations = false;
+    const varControl = <FormArray>this.newLimForm.get('variables');
+    for(let i = varControl.length-1; i >= 0; i--) {
+      varControl.removeAt(i);
+    }
+  }
+
+  public editLimitation(lim,limIndex) {
+    this.editFormLimitations = true;
+    this.newLimForm.controls['criteria'].setValue(lim.criteria);
+    this.newLimForm.controls['description'].setValue(lim.description);
+    if (lim.id) { //can't add reggression region id or limitation id if theres no regression region yet
+      this.newLimForm.controls['regressionRegionID'].setValue(this.selectedRegRegion.id);
+      this.newLimForm.controls['id'].setValue(lim.id);
+    } else { //adding lim index in order to edit limitation w/o regression region
+      this.newLimForm.controls['limIndex'].setValue(limIndex); 
+    }
+    lim.variables.forEach((v, varIndex) => {
+      this.addVariable();
+      const control = <FormArray>this.newLimForm.get('variables');
+      control.controls[varIndex].get('id').setValue(v.id);
+      control.controls[varIndex].get('limitationID').setValue(v.limitationID);
+      this.unitTypes.forEach((unit,x) => {  
+        if (unit.id.toString() == v.unitTypeID.toString()){
+          control.controls[varIndex].get('unitTypeID').setValue(this.unitTypes[x].id);
+        }
+      });
+      this.variables.forEach((variable,x) => {  
+        if (variable.id.toString() == v.variableTypeID.toString()){
+          control.controls[varIndex].get('variableTypeID').setValue(this.variables[x].id);
+        }
+      });
+    });
```

Comment:
Is this how we update items in the add scenario modal? I wonder if there's a better way to do this by linking the html directly with it? If it looks like a long task, we could maybe add it to #206 (which I haven't looked at in a while)